src/Context.php

```diff
@@ -119,6 +119,10 @@ public function addSymbol($name)
      */
     public function addVariable(Variable $variable)
     {
+        if (!is_string($variable->getName())) {
```

Comment:
This check seems is needed to be in `Variable` -> `__construct` method
And throw `\InvalidArgumentException`